kotlin/find-address/src/main/java/com/esri/arcgisruntime/sample/findaddress/MainActivity.kt

```diff
@@ -0,0 +1,342 @@
+              else -> throw e
+            }
+          }
+        }
+      } else {
+        Log.i(TAG, "Trying to reload locator task")
+        locatorTask.retryLoadAsync()
+      }
+    }
+    locatorTask.loadAsync()
+  }
+
+  /**
+   * Turns a GeocodeResult into a Point and adds it to a graphic overlay on the map.
+   *
+   * @param geocodeResult a single geocode result
+   */
+  private fun displaySearchResultOnMap(geocodeResult: GeocodeResult) {
+    // clear graphics overlay of existing graphics
+    graphicsOverlay.graphics.clear()
+    // create graphic object for resulting location
+    val resultPoint = geocodeResult.displayLocation
+    val resultLocationGraphic = Graphic(resultPoint, geocodeResult.attributes, pinSourceSymbol)
+    // add graphic to location layer
+    graphicsOverlay.graphics.add(resultLocationGraphic)
+    mapView.setViewpointAsync(Viewpoint(geocodeResult.extent), 1f)
+    showCallout(resultLocationGraphic)
+  }
+
+  /**
+   * Identifies and shows a call out on a tapped graphic.
+   *
+   * @param motionEvent the motion event containing a tapped screen point
+   */
+  private fun identifyGraphic(motionEvent: MotionEvent) {
+    // get the screen point
+    val screenPoint: android.graphics.Point = android.graphics.Point(
+      motionEvent.x.roundToInt(), motionEvent.y.roundToInt()
+    )
+    // from the graphics overlay, get the graphics near the tapped location
+    val identifyResultsFuture: ListenableFuture<IdentifyGraphicsOverlayResult> =
+      mapView.identifyGraphicsOverlayAsync(graphicsOverlay, screenPoint, 10.0, false)
+    identifyResultsFuture.addDoneListener{
+      try {
+        val identifyGraphicsOverlayResult: IdentifyGraphicsOverlayResult = identifyResultsFuture.get()
+        val graphics = identifyGraphicsOverlayResult.graphics
+        // get the first graphic identified
+        if (graphics.size > 0) {
+          val identifiedGraphic: Graphic = graphics[0]
+          // show the callout of the identified graphic
+          showCallout(identifiedGraphic)
+        } else {
+          // dismiss the callout if no graphic is identified (e.g. tapping away from the graphic)
+          callout?.dismiss()
+        }
+      } catch (e: Exception) {
+        Log.e(TAG, "Identify error: " + e.message)
+      }
+    }
+  }
+
+  /**
+   * Shows the given graphic's attributes as a call out.
+   *
+   * @param graphic the graphic containing the attributes to be displayed
+   */
+  private fun showCallout(graphic: Graphic) {
+    // create a text view for the callout
+    val calloutContent = TextView(applicationContext).apply {
+      setTextColor(Color.BLACK)
+      // get the graphic attributes for place name and street address, and display them as text in the callout
+      this.text = graphic.attributes["PlaceName"].toString() + "\n" +
+          graphic.attributes["StAddr"].toString()
+    }
+
+    // configure the callout
+    val calloutStyle = Callout.Style(this, R.xml.callout_style)
```

Comment:
this line doesn't compile for me

Reply:
whoops! I'd been fiddling around with callout styles before you helped me with the best way to do it. forgot to take this out!